### PR TITLE
Dev/took 31

### PR DESCRIPTION
**Pull request by Osman-Fatih-Cakir**

### CODE REVIEW

File: Editor/Mod.cpp [hunk 1]
```diff
@@ -231,6 +231,11 @@ namespace ToolKit
 
     void ActionManager::GroupLastActions(int n)
     {
+      if (n == 0)
```
Comment (afraidofdark) [inline Editor/Mod.cpp:234]: why code gets here ? are we hiding a problem ?

File: Editor/Mod.cpp [hunk 2]
```diff
@@ -832,11 +837,11 @@ namespace ToolKit
     {
       Window::Type activeType = g_app->GetActiveWindow()->GetType();
       if // Stop text edit deletes to remove entities.
-      (
-        activeType != Window::Type::Viewport 
-        && activeType != Window::Type::Viewport2d
-        && activeType != Window::Type::Outliner
-      )
+        (
```
Comment (afraidofdark) [inline Editor/Mod.cpp:840]: fix indentation please

File: Editor/UI.cpp [hunk 3]
```diff
@@ -65,6 +65,8 @@ namespace ToolKit
     TexturePtr UI::m_lockIcon;
     TexturePtr UI::m_visibleIcon;
     TexturePtr UI::m_invisibleIcon;
+    TexturePtr UI::m_lockedIcon;
```
Comment (afraidofdark) [inline Editor/UI.cpp:68]: there is a lock icon already, why not use it ? If they are identical use only one icon please.

File: Editor/UI.cpp [hunk 4]
```diff
@@ -92,7 +94,7 @@ namespace ToolKit
         0x015f, 0x015f,
         0x00dc, 0x00dc,
         0x00fc, 0x00fc,
-        0 
+        0
```
Comment (afraidofdark) [inline Editor/UI.cpp:97]: why is there 0 here ?

File: Editor/UI.cpp [hunk 5]
```diff
@@ -988,12 +994,12 @@ namespace ToolKit
 
       if ((rightClick || leftClick || middleClick) && m_mouseHover) // Activate with any click.
       {
-        if 
-        (
-          ImGui::IsMouseDragging(ImGuiMouseButton_Left) ||
-          ImGui::IsMouseDragging(ImGuiMouseButton_Right) ||
-          ImGui::IsMouseDragging(ImGuiMouseButton_Middle)
-        )
+        if
```
Comment (afraidofdark) [inline Editor/UI.cpp:997]: fix this and fallowing indentation problems. Did you auto indent ? and are you using 2 space for tabs.

**Review by afraidofdark — APPROVED** [review]

**Review by afraidofdark — CHANGES_REQUESTED** [review]
there are thins you need to do mostly indentation problems

File: Editor/Mod.cpp (hunk 1 above)
Comment (Osman-Fatih-Cakir) [reply Editor/Mod.cpp:234]: Since there is no actions to take (n=0) no need go. Also below code crashes when n is 0.

**Review by afraidofdark — APPROVED** [review]

**Pull request merged**